src/main/java/org/mskcc/cbio/portal/util/GeneticProfileReader.java

```diff
@@ -83,10 +86,18 @@ public static GeneticProfile loadGeneticProfile(File file) throws IOException, D
                                           + existingGeneticProfile.getStableId() + ") but different genetic alteration type: "
                                           + existingGeneticProfile.getGeneticProfileId());
             }
-            if (!existingGeneticProfile.getDatatype().equals(geneticProfile.getDatatype())) {
-                throw new IllegalStateException("genetic_profile record found with same Stable ID ("
-                        + existingGeneticProfile.getStableId() + ") but different data type: "
-                        + existingGeneticProfile.getDatatype());
+            if (DISCRETE_LONG.name().equals(geneticProfile.getDatatype())) {
+                if (!Set.of(DISCRETE_LONG.name(), DISCRETE.name()).contains(existingGeneticProfile.getDatatype())) {
+                    throw new IllegalStateException("genetic_profile record found with same Stable ID ("
+                            + existingGeneticProfile.getStableId() + ") but unsupported data type: "
+                            + existingGeneticProfile.getDatatype());
+                }
+            } else {
+                if (!existingGeneticProfile.getDatatype().equals(geneticProfile.getDatatype())) {
+                    throw new IllegalStateException("genetic_profile record found with same Stable ID ("
+                            + existingGeneticProfile.getStableId() + ") but different data type: "
+                            + existingGeneticProfile.getDatatype());
+                }
```

Comment:
what is supported or not is checked elsewhere, no? Maybe here it would suffice to just have the last check?

Reply:
I think it's unfortunate how `DISCRETE_LONG` has been implemented.

`DISCRETE_LONG` is not a separate data type, but an input file shape. We have to remember that `DISCRETE_LONG` and `DISCRETE` are the same data types at the end while doing this validation.